File: Simulation/bj_5073.py
# 5073 삼각형과 세 변

if __name__ == "__main__":
  a, b, c = -1, -1, -1

  while True:
    
    a, b, c = map(int, input().split())
    max_num = max(a, b, c)

    if a==0 and b==0 and c==0:
      break
      
    if max_num >= (a+b+c-max_num):
      print("Invalid")
      
    elif a==b and b==c and c==a:
      print("Equilateral")
  
    elif a==b or b==c or c==a:
      print("Isosceles")
  
    elif a!=b and b!=c and c!=a:
      print("Scalene")
  
    

File: Math/bj_5073.cpp
#include <iostream>
#include <vector>
#include <algorithm>

int main() {
  std::vector<std::string> result_list;

  while (1) {
    int a, b, c;
    std::cin >> a >> b >> c;
    int max_val = std::max({ a, b, c });

    if (a == 0 && b == 0 && c == 0)
      break;

    if (a == b && b == c && c==a)
      result_list.push_back("Equilateral");
    else if(a+b+c - max_val <= max_val)
      result_list.push_back("Invalid");
    else if (a == b or b == c or c == a)
      result_list.push_back("Isosceles");
    else if (a != b != c)
      result_list.push_back("Scalene");
  }
  for (std::string result : result_list)
    std::cout << result << std::endl;
  return 0;
}